### ISSUE TITLE
Possible issue with serde feature checks

### ISSUE THREAD

**Issue by 06chaynes** [kornelski/rusty-http-cache-semantics#9]
I think there may be an issue enabling the serde feature in the 2.0.0 release. I noticed errors after upgrading explaining that the CachePolicy struct was missing the Deserialize and Serialize traits. After a bit of looking I think I've identified the issue. 

It seems the feature checks are expecting a feature named `with_serde` to be enabled: 
```
#[cfg_attr(feature = "with_serde" ...
```

But in Cargo.toml the feature looks to just be named `serde`:
```
[features]
default = ["serde"]
serde = ["dep:serde", "dep:http-serde"]
```

So I think the feature would need to be renamed to `with_serde` or the checks updated to look for `serde`.

**Comment by 06chaynes** [comment]
Opened a PR just in case you are satisfied with the solution of changing the feature name

**Comment by kornelski** [comment]
Thanks! Fixed in 2.0.1
